### PR TITLE
Fixed management of the indices in ProbabilisticEventBased

### PR DESCRIPTION
**Pull request by micheles**
Closes https://github.com/gem/oq-risklib/issues/807. The tests are running here: https://ci.openquake.org/job/zdevel_oq-risklib/1539

### CODE REVIEW

File: openquake/risklib/scientific.py [hunk 1]
```diff
@@ -207,32 +207,35 @@ def interpolate(self, gmvs):
...
 
-    def apply_to(self, means, covs, idxs, epsilons):
+    def sample(self, means, covs, idxs, epsilons):
         """
-        Apply the vulnerability function to a ground motion vector, by using
-        an epsilon vector of the sample length.
+        Sample the epsilons and applies the corrections to the means.
+        This method is called only if there are nonzero covs.
 
         :param gmvs:
```
Comment (ptormene) [inline openquake/risklib/scientific.py]: Please make the parameter name consistent in the signature and in the docstring

File: openquake/risklib/scientific.py [hunk 2]
```diff
@@ -450,25 +453,24 @@ def interpolate(self, gmvs):
...
-        value(s) on the curve.
-
-        Input IML value(s) is/are clipped to IML range defined for this
-        vulnerability function.
-
-        :param float array iml: IML value
-
-        :returns: :py:class:`numpy.ndarray` containing a number of interpolated
-            values equal to the size of the input (1 or many)
+        Sample the epsilons and applies the corrections to the probabilities.
```
Comment (ptormene) [inline openquake/risklib/scientific.py]: `Sample the epsilons and apply...`

**Comment by ptormene** [comment]
Besides minor comments about docstrings, LGTM

File: openquake/risklib/scientific.py (hunk 1 above)
Comment (micheles) [inline openquake/risklib/scientific.py]: Done

File: openquake/risklib/scientific.py (hunk 2 above)
Comment (micheles) [inline openquake/risklib/scientific.py]: Done

**Pull request merged**